File: src/main/java/in/dragonbra/javasteam/steam/handlers/steamapps/callback/VACStatusCallback.java
package in.dragonbra.javasteam.steam.handlers.steamapps.callback;

import in.dragonbra.javasteam.generated.MsgClientVACBanStatus;
import in.dragonbra.javasteam.steam.steamclient.callbackmgr.CallbackMsg;
import in.dragonbra.javasteam.util.stream.BinaryReader;

import java.io.ByteArrayInputStream;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

/**
 * This callback is fired when the client receives it's VAC banned status.
 */
public class VACStatusCallback extends CallbackMsg {

    private List<Integer> bannedApps;

    public VACStatusCallback(MsgClientVACBanStatus msg, byte[] payload) {
        List<Integer> tempList = new ArrayList<>();

        try (BinaryReader br = new BinaryReader(new ByteArrayInputStream(payload))) {
            for (int i = 0; i < msg.getNumBans(); i++) {
                tempList.add(br.readInt());
            }
        } catch (IOException e) {
            throw new IllegalArgumentException("failed to read bans", e);
        }

        bannedApps = Collections.unmodifiableList(tempList);
    }

    public List<Integer> getBannedApps() {
        return bannedApps;
    }
}

File: src/main/java/in/dragonbra/javasteam/steam/handlers/steamapps/callback/VACStatusCallback.kt
package `in`.dragonbra.javasteam.steam.handlers.steamapps.callback

import `in`.dragonbra.javasteam.base.ClientMsg
import `in`.dragonbra.javasteam.base.IPacketMsg
import `in`.dragonbra.javasteam.generated.MsgClientVACBanStatus
import `in`.dragonbra.javasteam.steam.steamclient.callbackmgr.CallbackMsg
import `in`.dragonbra.javasteam.util.log.LogManager
import `in`.dragonbra.javasteam.util.stream.BinaryReader
import java.io.ByteArrayInputStream
import java.util.*

/**
 * This callback is fired when the client receives its VAC banned status.
 */
class VACStatusCallback(packetMsg: IPacketMsg) : CallbackMsg() {

    companion object {
        private val logger = LogManager.getLogger<VACStatusCallback>()
    }

    /**
     * Gets a list of VAC banned apps the client is banned from.
     */
    val bannedApps: List<Int>

    init {
        val vacStatus = ClientMsg(MsgClientVACBanStatus::class.java, packetMsg)
        val msg = vacStatus.body

        val tempList: MutableList<Int> = ArrayList()

        try {
            BinaryReader(ByteArrayInputStream(vacStatus.payload.toByteArray())).use { br ->
                for (i in 0 until msg.numBans) {
                    tempList.add(br.readInt())
                }
            }
        } catch (e: Exception) {
            logger.error("failed to read bans", e)
        }

        bannedApps = tempList.toList()
    }
}
